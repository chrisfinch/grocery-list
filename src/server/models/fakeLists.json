{
  "lists": [
    {
      "id": "1",
      "title": "Demo list",
      "created": 1564695374960,
      "items": [
        {
          "name": "beans",
          "purchased": false
        },
        {
          "name": "bread",
          "purchased": false
        },
        {
          "name": "broccoli",
          "purchased": false
        },
        {
          "name": "cereal",
          "purchased": false
        },
        {
          "name": "squash",
          "purchased": false
        }
      ]
    },
    {
      "id": "2",
      "title": "Other list",
      "created": 1564695541598,
      "items": [
        {
          "name": "pasta",
          "purchased": false
        }
      ]
    }
  ]
}