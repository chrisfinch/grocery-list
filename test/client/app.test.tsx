import React from 'react';
import { shallow } from 'enzyme';
import App from '../../src/client/app';
import { lists as fakeLists } from './../fakeLists.json';

describe('App', () => {
    beforeEach(() => {
        fetchMock.resetMocks();
    });

    it('Renders Correctly', () => {
        fetchMock.mockResponses(
            [JSON.stringify(fakeLists), { status: 200 }],
            [JSON.stringify(fakeLists[0]), { status: 200 }]
        );

        const tree = shallow(<App />);

        expect(tree).toMatchSnapshot();
        expect(fetchMock.mock.calls.length).toEqual(1);
        expect(fetchMock.mock.calls[0][0]).toEqual('/lists');
    });
});
