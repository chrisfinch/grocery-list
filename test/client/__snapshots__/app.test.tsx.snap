// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders Correctly 1`] = `
<div>
  <h1>
    Grocery List
  </h1>
  <h2>
    Lists:
  </h2>
  <ul />
  <hr />
  <form
    action="#"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="New list name"
      type="text"
      value=""
    />
    <input
      type="submit"
      value="Create list"
    />
  </form>
  <hr />
  <ListView
    activeListId={null}
  />
</div>
`;
